### ISSUE TITLE
`isEqualToComparingFieldByField` cannot be replaced with `usingRecursiveAssertion().isEqualTo`

### ISSUE THREAD

**Issue by filiphr** [assertj/assertj#3143]
**Describe the bug**
As the title says if I replace the deprecated `isEqualToComparingFieldByField` with the suggested solution `usingRecursiveAssertion().isEqualTo`, then my tests start to fail. See the provided test case

* assertj core version: 3.24.4
* java version: 17
* test framework version: JUnit Juputer 5.9.3
* os (if relevant): not relevant

**Test case reproducing the bug**

```java
public class ExampleTest {

    @Test
    void isEqualToComparingFieldByField() {
        // Passes
        assertThat(new Customer("Tester"))
                .isEqualToComparingFieldByField(new Customer("Tester"));
    }

    @Test
    void usingRecursiveAssertion() {
        // Fails
        assertThat(new Customer("Tester"))
                .usingRecursiveAssertion()
                .isEqualTo(new Customer("Tester"));
    }

    public static class Customer {

        private final String name;

        public Customer(String name) {
            this.name = name;
        }

        public String getName() {
            return name;
        }
    }

}
```

**Comment by scordio** [comment]
You're absolutely right because [`isEqualToComparingFieldByField`](https://www.javadoc.io/doc/org.assertj/assertj-core/latest/org/assertj/core/api/AbstractObjectAssert.html#isEqualToComparingFieldByField(java.lang.Object)) should be replaced by [`usingRecursiveComparison`](https://www.javadoc.io/static/org.assertj/assertj-core/3.24.2/org/assertj/core/api/AbstractObjectAssert.html#usingRecursiveComparison())

😉

**Comment by scordio** [comment]
This issue highlights a weakness of the new recursive assertion API: there are assertions pulled from `AbstractAssert` like `isEqualTo` which are not applied recursively and this is not obvious to the end user.

**Comment by filiphr** [comment]
OK this is really stupid from my side, I used `usingRecursiveAssert` instead of `usingRecursiveComparison`.  I started typing `usingRecur` and I picked the first one, didn't really think about it to be honest. 

Why can't the cant the `usingRecursiveAssert` and `usingRecursiveComparison` be merged?

**Comment by joel-costigliola** [comment]
recursive assertion is  a late addition, it can't be merged with the recursive comparison as it serves a different purpose than the recursive comparison, more like: performs this assertion on the each node of the object graph under test while the recursive comparison compares two graphs of objects.

**Comment by JitendraProgrammer** [comment]
This should pass in this case

import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.Test;

public class ExampleTest {

    @Test
    void isEqualToComparingFieldByField() {
        // Passes
        Assertions.assertThat(new Customer("Tester"))
                .isEqualToComparingFieldByField(new Customer("Tester"));
    }

    @Test
    void usingRecursiveAssertion() {
        // Should now pass
        Assertions.assertThat(new Customer("Tester"))
                .usingRecursiveComparison()
                .isEqualTo(new Customer("Tester"));
    }

    public static class Customer {

        private final String name;

        public Customer(String name) {
            this.name = name;
        }

        public String getName() {
            return name;
        }

        @Override
        public boolean equals(Object o) {
            if (this == o) return true;
            if (o == null || getClass() != o.getClass()) return false;
            Customer customer = (Customer) o;
            return name.equals(customer.name);
        }

        @Override
        public int hashCode() {
            return name.hashCode();
        }
    }
}

**Comment by joel-costigliola** [comment]
@JitendraProgrammer thanks for wanting to help, please try to analyze the issue a bit more before making a suggestion :wink:
